src/vcpkg/sourceparagraph.cpp

```diff
@@ -692,27 +687,21 @@ namespace vcpkg
             r.optional_object_field(obj, DEFAULT_FEATURES, dep.default_features, Json::BooleanDeserializer::instance);
             r.optional_object_field(obj, HOST, dep.host, Json::BooleanDeserializer::instance);
             r.optional_object_field(obj, PLATFORM, dep.platform, PlatformExprDeserializer::instance);
+
+            std::string raw_version_ge_text;
             auto has_ge_constraint = r.optional_object_field(
-                obj, VERSION_GE, dep.constraint.value, VersionConstraintStringDeserializer::instance);
+                obj, VERSION_GE, raw_version_ge_text, VersionConstraintStringDeserializer::instance);
             if (has_ge_constraint)
             {
                 dep.constraint.type = VersionConstraintKind::Minimum;
-                const auto& constraint_value = dep.constraint.value;
-                auto h = constraint_value.find('#');
-                if (h != std::string::npos)
+                auto maybe_version = Version::parse(raw_version_ge_text);
+                if (auto version = maybe_version.get())
                 {
-                    auto opt = Strings::strto<int>(ZStringView{constraint_value}.substr(h + 1));
-                    auto v = opt.get();
-                    if (v && *v >= 0)
-                    {
-                        dep.constraint.port_version = *v;
-                    }
-                    else
-                    {
-                        r.add_generic_error(type_name(),
-                                            msg::format(msgVersionConstraintPortVersionMustBePositiveInteger));
-                    }
-                    dep.constraint.value.erase(h);
+                    dep.constraint.version = std::move(*version);
+                }
+                else
+                {
+                    r.add_generic_error(type_name(), msg::format(msgVersionConstraintPortVersionMustBePositiveInteger));
```

Comment:
Is this the only reason that `Version::parse()` could fail? Choosing the specific error message here seems a bit far from the source.